codegen/module_system.go

```diff
@@ -1098,3 +1116,46 @@ func GenerateDependencyStruct(
 		packageHelper,
 	)
 }
+
+// sort the clients by client height in DAG, lowest node first
```

Comment:
How do you deal with cycles? We need to make sure we clearly surface any cycles in the DAG rather than relying on compilation errors.